consensus/src/sync/live/block_queue/queue.rs

```diff
@@ -833,6 +833,27 @@ impl<N: Network> Stream for BlockQueue<N> {
                         return Poll::Ready(Some(block));
                     }
                 }
+                Poll::Ready(Some(Err(syn_queue_error))) => {
+                    // The request failed, remove from the buffer all blocks from the target block chain that errored.
+                    // Do this by locating the successor of the block if it exists and see if there is a block whose
+                    // parent hash is the block in question.
+                    let mut block_number = syn_queue_error.target_block_number;
+                    let mut block_hash = syn_queue_error.target_block_hash;
+                    while let Some(blocks) = self.buffer.get_mut(&(block_number + 1)) {
+                        let Some((hash, (block, _source))) = blocks
+                            .iter()
+                            .find(|(_hash, (block, _source))| *block.parent_hash() == block_hash)
+                        else {
+                            // No block found whose parent block is the one we're looking: we should have already removed
+                            // all blocks from the target block chain that errored.
+                            break;
+                        };
+                        block_number = block.block_number();
+                        block_hash = hash.clone();
+                        blocks.remove(&block_hash);
+                    }
+                    return Poll::Ready(None);
```

Comment:
I think this is incorrect. The block queue should not return None since that means it has terminated by rust stream rules. Here we should instead do nothing and let it loop again so that it registers the weaker or returns the next value.